packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -533,6 +533,8 @@ export class ConnectionManager implements IConnectionManager {
 		const docService = this.serviceProvider();
 		assert(docService !== undefined, 0x2a7 /* "Container is not attached" */);
 
+		this.props.establishConnectionHandler(reason);
```

Comment:
It's minor but I wonder if establishingconnection state should always be equivalent to pendingConnection being set.

In terms of the change, I think you could just move the StorageOnly case down lower rather than moving this event emit up. I'm on my phone so didn't check but I assume setupNewSuccessfulConnection will take care of the pending connection.

Reply:
@markfields - I'll glance a bit later at it, I want to unblock the release, so I'll merge as is and if anything, to change, I'll incorporate it in the PR that I mentioned in description. (Sorry, need to leave in a minute and release is today)